force-app/main/default/classes/Homework.cls

```diff
@@ -115,6 +135,8 @@ public with sharing class Homework
 
         // Do not use an if statement!
 
+        result == (value <upper) && (value>lower);
```

Comment:
Close, but you've used `==` which is a comparison operator. You want assignment which is `=` (single equal rather than two).